runtime/citrus-groovy/src/main/java/com/consol/citrus/groovy/dsl/GroovyShellUtils.java

```diff
@@ -0,0 +1,113 @@
+import java.util.regex.Pattern;
+
+import com.consol.citrus.Citrus;
+import com.consol.citrus.context.TestContext;
+import groovy.lang.Binding;
+import groovy.lang.GroovyShell;
+import groovy.lang.Script;
+import org.codehaus.groovy.control.CompilerConfiguration;
+import org.codehaus.groovy.control.customizers.ImportCustomizer;
+
+/**
+ * @author Christoph Deppisch
+ */
+public class GroovyShellUtils {
+
+    private static final Pattern COMMENTS = Pattern.compile("^(?:\\s*//|/\\*|\\s+\\*).*$", Pattern.MULTILINE);
+
+    private GroovyShellUtils() {
+        // prevent instantiation of utility class
+    }
+
+    /**
+     * Run given scriptCode with GroovyShell.
+     * @param ic import customizer
+     * @param scriptCode code to evaluate in shell
+     * @param context the current test context
+     * @param <T> return type
+     * @return script result
+     */
+    public static <T> T run(ImportCustomizer ic, String scriptCode, Citrus citrus, TestContext context) {
+        return run(ic, null, scriptCode, citrus, context);
+    }
+
+    /**
+     * Run given scriptCode with GroovyShell and delegate execution to given instance.
+     * @param delegate instance providing methods and properties
+     * @param scriptCode code to evaluate in shell
+     * @param context the current test context
+     * @param <T> return type
+     * @return script result
+     */
+    public static <T> T run(Object delegate, String scriptCode, Citrus citrus, TestContext context) {
+        return run(new ImportCustomizer(), delegate, scriptCode, citrus, context);
+    }
+
+    /**
+     * Run given scriptCode with GroovyShell and delegate execution to given instance.
+     * @param ic import customizer
+     * @param delegate instance providing methods and properties
+     * @param scriptCode code to evaluate in shell
+     * @param context the current test context
+     * @param <T> return type
+     * @return script result
+     */
+    public static <T> T run(ImportCustomizer ic, Object delegate, String scriptCode, Citrus citrus, TestContext context) {
+        CompilerConfiguration cc = new CompilerConfiguration();
+        cc.addCompilationCustomizers(ic);
+        cc.setScriptBaseClass(GroovyScript.class.getName());
+
+        ClassLoader cl = Thread.currentThread().getContextClassLoader();
+        GroovyShell sh = new GroovyShell(cl, new Binding(), cc);
+
+        Script script = sh.parse(scriptCode);
+
+        if (script instanceof GroovyScript) {
+            if (delegate != null) {
+                // set the delegate target
+                ((GroovyScript) script).setDelegate(delegate);
+            }
+
+            ((GroovyScript) script).setCitrusFramework(citrus);
+            ((GroovyScript) script).setContext(context);
+        }
+
+        return (T) script.run();
+    }
+
+    /**
+     * Remove leading comments such as license header.
+     * @param script
+     * @return
+     */
+    public static String removeComments(String script) {
```

Comment:
why is this important? shouldn't comments be ignored anyway?

Reply:
removing the leading comments makes it easier to identify the script nature in `isActionScript()` which uses startsWith() checks

Reply:
ok, I just thought it is probably also a bit hacky in a way. cannot really explain it, was just a feeling :D